### DISCUSSION TITLE
forge invariant test not fuzzing functions

### DISCUSSION THREAD

**Discussion by jaimebarrancos** [Cyfrin/security-and-auditing-full-course-s23#101] [Q&A] [answered]
I am writing invariant tests using a handler contract in foundry. However, for some reason, the handler contract is not fuzzing (I know because when I run `forge test --mt assertSomething` It doesn't show amount of runs / depth / anything fuzz related).

```javascript
contract Invariant is StdInvariant, Test {

    Handler handler;

    //additional logic ...

    function setUp() public {
        handler = new Handler(pool);

        //additional logic ...

        bytes4[] memory _selectors = new bytes4[](4);
        _selectors[0] = handler.deposit.selector;
        _selectors[1] = handler.swap.selector;

        targetSelector(FuzzSelector({ addr: address(handler), selectors: _selectors}));
        targetContract(address(handler));
    }

    function assertSomething() {
    //logic ...
}

```

Functions in handler contract:
```javascript
    //deposit
    function deposit(uint256 approvedAndDepositedToken, uint256 approvedAndDepositedWeth) public {
        //logic ...
    }

    //swap
    function swap(uint256 swapAmt) public {
    //logic ...
    }
```

This is my output:

[![output][1]][1]

Why are the functions in my handler contract not fuzzing (maybe not even running at all)?

  [1]: https://i.stack.imgur.com/Pu1V8.png

**Comment by jaimebarrancos** [comment, marked as answer]
I figured it out. The problem was the array length I specified was too large (**4**). The space of the array I specified is not the amount that I actually used.

> Foundry probably checks if any selector of the selectors array is empty / not used.

I had to change the code from:

```solidity
bytes4[] memory selectors = new bytes4[](4);
bytes4[] memory selectors = new bytes4[](2);
```

Because I only had 2 functions to fuzz call.

**Reply by PatrickAlphaC** [reply]
Way to figure it out!
